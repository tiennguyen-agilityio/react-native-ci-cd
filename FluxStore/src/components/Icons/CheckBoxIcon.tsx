import {memo} from 'react';
import {TouchableOpacity} from 'react-native';
import Svg, {Path, Rect} from 'react-native-svg';

import {IconProps} from '@/interfaces';
import {useThemeStore} from '@/stores';
import {colors} from '@/themes';

export const CheckBoxIcon = memo(
  ({
    width = 19,
    height = 20,
    color,
    disabled = false,
    isActive = false,
    style,
    onPress,
  }: IconProps) => {
    const {
      theme: {icon, text},
    } = useThemeStore();

    return (
      <TouchableOpacity
        testID="checkbox-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}>
        <Svg width={width} height={height} viewBox="0 0 19 20" fill="none">
          <Rect
            x={0.592896}
            width={18.2891}
            height={20}
            rx={4}
            fill={color ?? isActive ? colors.green[200] : icon.checkbox}
          />
          <Path
            d="M5.165 9.47L8.014 13l5.381-6"
            stroke={text.light}
            strokeWidth={2}
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </Svg>
      </TouchableOpacity>
    );
  },
);
