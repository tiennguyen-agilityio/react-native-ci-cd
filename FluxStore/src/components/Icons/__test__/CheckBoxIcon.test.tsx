import {render, fireEvent, screen} from '@testing-library/react-native';

import {CheckBoxIcon} from '../CheckBoxIcon';

describe('CheckBoxIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<CheckBoxIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with isActive true', () => {
    const {toJSON} = render(<CheckBoxIcon isActive />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<CheckBoxIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('checkbox-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<CheckBoxIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('checkbox-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
