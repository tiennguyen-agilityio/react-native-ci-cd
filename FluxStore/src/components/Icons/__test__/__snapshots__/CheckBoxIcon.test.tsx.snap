// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="checkbox-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={20}
    bbWidth={19}
    fill="none"
    focusable={false}
    height={20}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 20,
          "width": 19,
        },
      ]
    }
    vbHeight={20}
    vbWidth={19}
    width={19}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGRect
        fill={
          {
            "payload": 4287532686,
            "type": 0,
          }
        }
        height={20}
        propList={
          [
            "fill",
          ]
        }
        rx={4}
        width={18.2891}
        x={0.592896}
        y={0}
      />
      <RNSVGPath
        d="M5.165 9.47L8.014 13l5.381-6"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`CheckBoxIcon renders correctly (matches snapshot) with isActive true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="checkbox-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={20}
    bbWidth={19}
    fill="none"
    focusable={false}
    height={20}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 20,
          "width": 19,
        },
      ]
    }
    vbHeight={20}
    vbWidth={19}
    width={19}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGRect
        fill={
          {
            "payload": 4283468411,
            "type": 0,
          }
        }
        height={20}
        propList={
          [
            "fill",
          ]
        }
        rx={4}
        width={18.2891}
        x={0.592896}
        y={0}
      />
      <RNSVGPath
        d="M5.165 9.47L8.014 13l5.381-6"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4294967295,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={2}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
